apps/site/src/lib/admin/useApplicant.ts

```diff
@@ -0,0 +1,79 @@
+import axios from "axios";
+import useSWR from "swr";
+
+export type Uid = string;
+
+export enum Decision {
+	accepted = "ACCEPTED",
+	rejected = "REJECTED",
+	waitlisted = "WAITLISTED",
+}
+
+export type Review = [string, Uid, Decision];
+
+// The application responses submitted by an applicant
+export interface ApplicationData {
+	first_name: string;
+	last_name: string;
+	email: string;
+	pronouns: string[];
+	ethnicity: string;
+	is_18_older: boolean;
+	school: string;
+	education_level: string;
+	major: string;
+	is_first_hackathon: boolean;
+	portfolio: string | null;
+	linkedin: string | null;
+	frq_collaboration: string;
+	frq_dream_job: string;
+	resume_url: string;
+	submission_time: string;
+	reviews: Review[];
+}
+
+export type ApplicationQuestion = Exclude<keyof ApplicationData, "reviews">;
+
+export enum ReviewStatus {
+	pending = "PENDING_REVIEW",
+	reviewed = "REVIEWED",
+	released = "RELEASED",
+}
+
+export const Status = { ...ReviewStatus, ...Decision };
+export type Status = ReviewStatus | Decision;
+
+export interface Applicant {
+	_id: Uid;
+	role: string;
+	status: Status;
+	application_data: ApplicationData;
+}
+
+const fetcher = async ([api, uid]: [string, string]) => {
```

Comment:
Nitpick: Should we use the `Uid` type here instead of `string`?

Reply:
Used to be `string[]` before the type fix, but makes sense since we're switching to tuple type.